waltz-client/src/main/java/com/wepay/waltz/client/internal/Partition.java

```diff
@@ -564,6 +566,66 @@ private void sendTransactionDataRequest(long transactionId) {
         }
     }
 
+    /**
+     * Resubmits pending HighWaterMark requests.
+     */
+    public void resubmitHighWaterMarkRequests() {
+        if (highWaterMarkRef.get() != null) {
+            sendHighWaterMarkRequest();
+        }
+    }
+
+    /**
+     * Invoked when the high watermark is received from a Waltz server.
+     *
+     * @param highWaterMark the high watermark.
+     */
+    public void highWaterMarkReceived(long highWaterMark) {
+        while (true) {
```

Comment:
No need to do `while` loop. If `highWaterMarkRef.get()` returns `null`, this will keep looping until someone requests a high-water mark.

Reply:
You are right! Update it now!